### DISCUSSION TITLE
Best way to track that signals have sent from examples for perfect negotiation

### DISCUSSION THREAD

**Discussion by srcrip** [elixir-webrtc/ex_webrtc#93] [Q&A]
Hi,

I'm trying to implement perfect negotiation, based off the echo examples. In the ws handler, how can I find out that my ws message has actually sent? Because for purposes of perfect negotiation you generally want to track when an offer has finished going out? Maybe I'm over thinking it, I hope this kinda makes sense.

**Comment by mickel8** [comment]
Hi @srcrip,

>  how can I find out that my ws message has actually sent

sent or received?


In general, I would recommend taking a look at https://blog.mozilla.org/webrtc/perfect-negotiation-in-webrtc/

We don't support rollbacks right now so it's not possible to implement perfect negotiation in a way described in the article. 

Right now, you need to make sure, using your own signaling, that only one side can generate an offer at the time.

**Comment by LVala** [comment]
Closing as inactive.
